### PR TITLE
[CSL-854] Add tests to unsuccessful verification

### PR DESCRIPTION
**Pull request by rockbmb**
Tests that verify successfuly verification of correct 'Commitment's
and 'Opening's already existed. This commit now adds tests to verify
incorrect values of these types will not be validated.

### CODE REVIEW

**Comment by volhovM** [comment]
I don't think that "incorrect arbitrary" should be in main library. Is there any way to move them to `test` folder?

**Comment by gromakovsky** [comment]
Well, later ssc will be moved into a separate package. We will have some tests there I suppose. Some of them maybe use incorrect arbitrary. Also later (in 2019) we will have some global tests in top-level package. There we can also need incorrect arbitrary.
Alternatively we can put all tests into top-level package.

File: src/Pos/Ssc/GodTossing/Arbitrary.hs [hunk 1]
```diff
@@ -39,12 +42,54 @@ import           Pos.Util.Arbitrary               (Nonrepeating (..), makeSmall,
...
+-- | Wrapper over 'Commitment'. Creates an invalid Commitment w.r.t. 'verifyCommitment'.
+newtype BadCommitment = BadComm
+    { getBadComm :: Commitment
+    } deriving (Show, Eq)
+
+instance Arbitrary BadCommitment where
+    arbitrary = BadComm <$> do
+      Commitment <$> arbitrary <*> arbitrary <*> (arbitrary `suchThat` (not . null))
+
+-- | Wrapper over 'Opening'. Creates an invalid Opening w.r.t. 'verifyOpening'.
```
Comment (gromakovsky) [inline src/Pos/Ssc/GodTossing/Arbitrary.hs]: I don't understand this wrapper.
1. `Opening` itself can't be valid or invalid. Validity of `Opening` can be determined only when we know a `Commitment`.
2. As a consequence, `BadOpening`'s `Arbitrary` doesn't have any logic.

I think this wrapper is redundant.

**Comment by rockbmb** [comment]
@volhovM There is, the growing number of types created only for testing is making me seriously consider moving all of them into their respective modules. I'll try to get some consensus about it, and then if there are people in favor, I'll do it.

File: src/Pos/Ssc/GodTossing/Arbitrary.hs [hunk 2]
```diff
@@ -39,12 +42,46 @@ import           Pos.Util.Arbitrary               (Nonrepeating (..), makeSmall,
...
 ----------------------------------------------------------------------------
 
+-- | Wrapper over 'Commitment'. Creates an invalid Commitment w.r.t. 'verifyCommitment'.
+newtype BadCommitment = BadComm
+    { getBadComm :: Commitment
+    } deriving (Show, Eq)
+
+instance Arbitrary BadCommitment where
+    arbitrary = BadComm <$> do
+      Commitment <$> arbitrary <*> arbitrary <*> (arbitrary `suchThat` (not . null))
```
Comment (gromakovsky) [inline src/Pos/Ssc/GodTossing/Arbitrary.hs]: Is this `suchThat` necessary here? 😕

Comment (rockbmb) [inline src/Pos/Ssc/GodTossing/Arbitrary.hs]: Oh, I forgot to remove this part...
It's related to the edge case I mentioned about `verifyCommitment`. I'll remove this now.

**Pull request merged**